engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -944,6 +944,27 @@ public static boolean isTypedVector(Class<?> type) {
     private Expression[] convertParameters(final Executable executable,
             final Class<?>[] argumentTypes, final Class<?>[] expressionTypes,
             final Class<?>[][] parameterizedTypes, Expression[] expressions) {
+
+        // For Python callables, all Vector columns should be converted into arrays
```

Comment:
I have a few questions / comments after reading this:
1) This looks like it converts all vectors to arrays, right?
2) I can see doing the array conversion as being convenient, but I can also see it as being low performance.  Do we understand the tradeoffs of always doing it vs having a method in python that gets called?
3) This looks like nodes are being added to the query call expressions.  Will this change effectively break vectorization for python until we do extensive work to make the rest of vectorization work?
4) Have any tests been run to quantify (3)?

Reply:
1. yes
2. it is actually more performant because JPY knows how to handle Java arrays not vectors. So converting vectors to arrays first should reduce the boundary-crossing when the Python code iterates over the data.
3. Python function calls with array/vector argument can't be vectorized.

Reply:
Does the python code see the input as a numpy array or just a buffer that needs to be converted to a numpy array?

Reply:
The Python code sees it as a JPY wrapped Java array not a numpy array. If the array is a primitive one, its buffer can be used to create a numpy array with zero-copy.

Reply:
This seems very usable without always converting to numpy.

![image](https://user-images.githubusercontent.com/5250374/221966410-2a09f493-2d1d-454c-9733-ba8c7727583b.png)

My only concern is the lack of vectorization, but I think that is a separate effort.